### PR TITLE
Add Gitpod.io integration

### PR DESCRIPTION
**Pull request by theodorosploumis**
| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (use the 4.x branch)      | [x]
| Related user documentation PR URL      | _TBD_

#### Description:

Since we have ddev support it would be nice to also have a direct https://gitpod.io integration that creates a "managed" Mautic instance.

In this PR I have also create a custom ddev command which is a wrapper of the `.ddev/mautic-setup.sh` script. Saying that, I would suggest to also move the whole script `.ddev/mautic-setup.sh` into ddev commands.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10500"></a>

### CODE REVIEW

**Comment by RCheesley** [comment]
Thanks for the PR @theodorosploumis - Gitpod looks really interesting! I've asked @dennisameling to review as he wrote all of the DDEV integration side of things, so he's probably the best person to provide feedback.

If you could sign the Contributors Agreement that would be awesome :)

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
We can disregard the code coverage check in this case because you are not making changes to any of the tracked files, so we're good to proceed with testing :)

File: .gitpod.Dockerfile
Comment (dennisameling) [inline .gitpod.Dockerfile]: I don't think we need PhpStorm in here as Gitpod has built-in VS Code support, right?

File: .gitpod.yml
Comment (dennisameling) [inline .gitpod.yml:26]: So if I understand correctly, folks will only be able to open Mautic itself, not MailHog and PHPMyAdmin, right? That's totally fine, just curious and might be nice for a future iteration 😊

File: .gitpod.yml
Comment (dennisameling) [inline .gitpod.yml]: Curious to see what this will do. [The official docs](https://www.gitpod.io/docs/references/gitpod-yml#prebuilds.addcheck) mention the following:

> Configure whether Gitpod registers itself as a status check to pull requests

Would this mean that, in case Gitpod fails to build for some reason, we'll have a failed check? That might be painful, as we already test things quite extensively directly in the GitHub Actions CI pipeline. But we can leave it on for now (it's true by default as well) and change it later if needed.

File: .gitpod.yml
Comment (dennisameling) [inline .gitpod.yml]: I'd rather not have Gitpod add a comment every time someone creates a PR. The `addBadge` setting below, which is set to true as well, will simply update the PR description with a button to open it in Gitpod. Think that's enough. What do you think @RCheesley?

**Review by dennisameling — COMMENTED** [review]
_Very_ nice initiative, thank you! 🚀 I've left some notes and questions.

File: .gitpod.yml
Comment (RCheesley) [reply .gitpod.yml:26]: Yeah I agree it would be nice if it was possible to have both Mailhog and PHPMyAdmin. The former is probably the most important from the testing perspective.

File: .gitpod.yml
Comment (RCheesley) [reply .gitpod.yml]: I think it would be helpful to have an instance for each PR from forks re. the first option - it could replace Mautibox perhaps from the testing perspective? 

If the build itself fails (eg something got broken with the installation process) we should pick that up with the other tests indeed - maybe @theodorosploumis could clarify what actually happens if we do have the check selected. I'd rather not have another status check unless it is going to tell us something we don't already know.

File: .gitpod.yml
Comment (RCheesley) [reply .gitpod.yml]: I think that adding a comment would add a lot of noise! If the button is clear enough then it should be sufficient.

**Comment by RCheesley** [comment]
Something to clarify, Gitpod is a paid service, so if we have this running on all our community PRs I presume we will need to look into a community account. 

If this were to replace Mautibox we do have an amount in the budget which we could use for that, so we would need some clarity on this. @theodorosploumis do you have any insights on how this would work for the project?

File: .gitpod.Dockerfile
Comment (theodorosploumis) [reply .gitpod.Dockerfile]: Yes, this was there by mistake from another official Dockerfile I used. I will remove it.

File: .gitpod.yml
Comment (theodorosploumis) [reply .gitpod.yml:26]: Both Mailhog (port 8025) and Phpmyadmin (port 8036) are  through DNS binding. Gitpod manages the ports and adds the port prefix in front of the main instance url. 
The comments here say that the ports are ignored when they bind to to Gitpod. "Ignore" word here is for Gitpod. Otherwise Gitpod will open the url for each port.

Notice that Gitpod only exposes the `http` ports and not the `https`.

In any case I will remove these comments accordingly so it is clear.

File: .gitpod.yml
Comment (theodorosploumis) [reply .gitpod.yml]: But all these settings only work if someone form the Team adds Gitpod integration with this repository through https://github.com/apps/gitpod-io. Otherwise the options `github` is just for reference.

See docs here: https://www.gitpod.io/docs/references/gitpod-yml#github

I am not sure about the `addCheck` option since I have not use it up to now. We have a pretty simple Dockerfile and the only check here is that Mautic can be installed through the `tasks` commands found on file `.gitpod.yml`. 

Notice, also, that I already use the code from this PR and I (pre)build the image up to now without issues. 
This saved me a lot of time when creating the Mautic instance.

In any case I will set the `addCheck` option to `FALSE` for now and we can come on this later.

**Comment by theodorosploumis** [comment]
> Something to clarify, Gitpod is a paid service, so if we have this running on all our community PRs I presume we will need to look into a community account.
> 
> If this were to replace Mautibox we do have an amount in the budget which we could use for that, so we would need some clarity on this. @theodorosploumis do you have any insights on how this would work for the project?

Gitpod has a special program for Open Source software https://www.gitpod.io/docs/professional-open-source. 
I am not sure if the Mautic Team is eligible for this but it seems we have many chances.

I just saw that there is already a **ddev integration** and then I wanted to create several instances online through Gitpod. I was making a large scale integration to Drupal 9.x and I needed to check and present so many things online as also as to run Mautic cron tasks through the console. 

Also, my indent is not to replace Mauticbox (which I didn't know up to know to be honest). But if Mauticbox has issues or limitations that would be an option.

In any case, Gitpod integration does not harm to just exist there. Both tools can exist on the same place.

But, for my work, Gitpod is not only a testing tool :-)

**Comment by RCheesley** [comment]
@theodorosploumis I was not fully sure how to test this so I just dropped gitpod.io in front of the PR's URL. 

It installed mautic but these links are not working:

screenshot-ivory-gazelle-u63rrh73 ws-eu17 gitpod io-2021 10 15-12_38_58

I got some notifications about ports being available and was asked if I wanted to make them public - even selecting yes did not help. Maybe we need to update the URLs?

**Comment by RCheesley** [comment]
OK so from my testing

✅  Mautic installed and working
✅  Mailhog installed and working - accessed at https://8025-example.gitpod.io/ via the remote icon on the left and clicking open preview
❌  PHPMyAdmin is not loading - 400 bad request when I try to access it on 8037: `400 Bad Request The plain HTTP request was sent to HTTPS port `- when I try to access on 8036 I just get a white screen. So something seems a bit broken there.

File: .gitpod.yml
Comment (theodorosploumis) [reply .gitpod.yml]: Removed this option.

**Comment by theodorosploumis** [comment]
> OK so from my testing
> 
> white_check_mark Mautic installed and working white_check_mark Mailhog installed and working - accessed at https://8025-example.gitpod.io/ via the remote icon on the left and clicking open preview x PHPMyAdmin is not loading - 400 bad request when I try to access it on 8037: `400 Bad Request The plain HTTP request was sent to HTTPS port `- when I try to access on 8036 I just get a white screen. So something seems a bit broken there.

@RCheesley  I can open the Phpmyadmin normally on port 8036 (https://8036-EXAMPLE.gitpod.io).
Did you check the services with `ddev describe`? Maybe Phpmyadmin service is not available for some reason.

**Comment by RCheesley** [comment]
Yeah I did - here's the screenshot and you can see at the top the white screen.

screenshot-coral-gibbon-l88o0mti ws-eu17 gitpod io-2021 10 15-13_17_14

**Comment by theodorosploumis** [comment]
@RCheesley can you please click the "Share" button on bottom left of Gitpod so I can check it online?
Otherwise, can you please make the port 8036 as "Public"?

**Comment by RCheesley** [comment]
For sure: https://coral-gibbon-l88o0mti.ws-eu17.gitpod.io

I made it public and it worked but not in the preview browser - whereas the mailhog and mautic apps work fine there. Maybe a quirk of the built in browser?

**Comment by theodorosploumis** [comment]
@RCheesley Yes. It seems that Phpmyadmin does not work inside the preview window. It works only as a separate page (https://8036-coral-gibbon-l88o0mti.ws-eu17.gitpod.io/). Maybe because it send login cookies.

File: .gitpod.yml
Comment (RCheesley) [reply .gitpod.yml]: I've installed the application for this repo so we should be good to go once this is merged.

**Comment by RCheesley** [comment]
I think the PHPMyAdmin side of things may be because it runs on non-ssl. I came across it when working on another project.

I have been in touch with Gitpod and they have said that most people will be able to test PRs without any problems in terms of limitations on their personal (free) account. If we have people who are working regularly with our community and they are hitting the limits we can apply for them to be added to our list of approved team members which would grant them a pro account free of charge.

From my perspective I think this is a great addition and stands the chance of helping us to really speed up testing.

My only concern is that the URLs which are output (I think by DDEV) which I mentioned in this comment: https://github.com/mautic/mautic/pull/10500#issuecomment-944230465 seem to be malformed. Could just be a case of dropping a space after the comma and prefixing the gitpod URLs with http/https as appropriate perhaps?

Would be awesome to get this merged in sooner rather than later!

**Review by mabumusa1 — COMMENTED** [review]
I tested this PR, Gitpod looks like a great solution for quickly testing. 
I managed to get the following running 
- Mautic
- Mailhog 
- PhpMyadmin
- Getting the code opened within VS Code

I think the note of @RCheesley https://github.com/mautic/mautic/pull/10500#issuecomment-944230465 is required

**Comment by dennisameling** [comment]
Working on updating the URLs now :) have found a solution for that

**Review by dennisameling — APPROVED** [review]
Just fixed the issue with the URLs as reported by @RCheesley 😊 Let's goooo 🚀 

```
All done! Here's some useful information:
🔒 The default login is admin/mautic
🌐 To open the Mautic instance, go to https://8080-peach-tahr-206ew237.ws-eu18.gitpod.io in your browser.
🌐 To open PHPMyAdmin for managing the database, go to https://8036-peach-tahr-206ew237.ws-eu18.gitpod.io in your browser.
🌐 To open MailHog for seeing all emails that Mautic sent, go to https://8025-peach-tahr-206ew237.ws-eu18.gitpod.io in your browser.
🚀 Run "ddev exec composer test" to run PHPUnit tests.
🚀 Run "ddev exec bin/console COMMAND" (like mautic:segments:update) to use the Mautic CLI. For an overview of all available CLI commands, go to https://mau.tc/cli
🔴 If you want to stop the instance, simply run "ddev stop".
```

**Comment by RCheesley** [comment]
Yassssssss! 🚀 Thank you @theodorosploumis for this work!

@all-contributors please add @theodorosploumis for code

**Pull request merged**
